CASMPET-7265 add node to management-nodes-rollout to restart goss-servers on worker nodes (#5503)
Cray-HPE/docs-csm: operations/validate_csm_health.md

Each section of this health check document provides links to relevant troubleshooting procedures. If additional help is needed, see
[CSM Troubleshooting Information](../troubleshooting/README.md).

> Before running health checks, restart `goss-servers` on all NCNs to ensure the correct tests are run on each node. This is necessary because of a timing issue on the nodes that has been fixed in CSM 1.6.1.

Reviewer: Looking at the health checks on this page, I think we should move this to the beginning of the Platform health checks section. This is for two reasons:

1. Those are the only Goss tests called on this page
2. Other pages link directly to that item, in which case they could miss this blurb at the top.

Author: Makes sense. I have moved it.

All platform health checks are expected to pass. Each check has been implemented as a [Goss](https://github.com/aelsabbahy/goss) test which reports a `PASS` or `FAIL`.

Before running health checks, restart `goss-servers` on all NCNs to ensure the correct tests are run on each node. This is necessary because of a timing issue on the nodes that has been fixed in CSM 1.6.1.

(`ncn-m#` or `pit#`) Restart `goss-servers`.

```bash

> ncn_nodes=$(grep -oP "(ncn-s\w+|ncn-m\w+|ncn-w\w+)" /etc/hosts | sort -u | tr -t '\n' ',')

Reviewer: Per our Slack discussion, the command to get the NCN node list should be split between the PIT scenario and the non-PIT scenario.

Author: I have added the PIT scenario.

All platform health checks are expected to pass. Each check has been implemented as a [Goss](https://github.com/aelsabbahy/goss) test which reports a `PASS` or `FAIL`.

> Before running health checks, restart `goss-servers` on all NCNs to ensure the correct tests are run on each node. This is necessary because of a timing issue on the nodes that has been fixed in CSM 1.6.1.

Reviewer 2: Same here.

Reviewer: ```suggestion
If the CSM version is 1.6.0 or lower, then before running health checks, restart the `goss-servers` service on all NCNs to ensure the correct tests are run on each node. This is necessary because of a timing issue that is fixed in CSM 1.6.1.

> Skip the `goss-servers` service restart if the CSM version is 1.6.1 or above.
> The restart will cause no harm if done on CSM 1.6.1 or higher, but it is unnecessary.
```

Author: I am leaving the tense as is so that it does not need to be changed later.

Run one of the two commands below depending on if it is being executed from `ncn-m001` or from the PIT node.

> 1. (`ncn-m001#`) Run the following from `ncn-m001` to restart `goss-servers`.
> 
>     ```bash
>     ncn_nodes=$(grep -oP "(ncn-s\w+|ncn-m\w+|ncn-w\w+)" /etc/hosts | sort -u | tr -t '\n' ',')
>     ncn_nodes=${ncn_nodes%,}
>     pdsh -S -b -w $ncn_nodes 'systemctl restart goss-servers'
>     ```
> 
> 1. (`pit#`) Run the following from the PIT node to restart `goss-servers`.

Reviewer: ```suggestion
- (`ncn-m001#`) Run the following from `ncn-m001` to restart `goss-servers`.

    ```bash
    ncn_nodes=$(grep -oP "(ncn-s\w+|ncn-m\w+|ncn-w\w+)" /etc/hosts | sort -u | tr -t '\n' ',')
    ncn_nodes=${ncn_nodes%,}
    pdsh -S -b -w $ncn_nodes 'systemctl restart goss-servers'
    ```

- (`pit#`) Run the following from the PIT node to restart `goss-servers`.
```

All platform health checks are expected to pass. Each check has been implemented as a [Goss](https://github.com/aelsabbahy/goss) test which reports a `PASS` or `FAIL`.

If the CSM version is 1.6.0 or lower, then before running health checks, restart the `goss-servers` service on all NCNs to ensure the correct tests are run on each node. This is necessary because of a timing issue that is fixed in CSM 1.6.1.

> Skip the `goss-servers` service restart if the CSM version is 1.6.1 or above.
> The restart will cause no harm if done on CSM 1.6.1 or higher, but it is unnecessary.

> Run one of the two commands below depending on if it is being executed from `ncn-m001` or from the PIT node.

Reviewer: We need this whitespace here, or else the "Run..." part will be part of the previous comment, which we do not want.

```suggestion

Run one of the two commands below depending on if it is being executed from `ncn-m001` or from the PIT node.
```
